orsopy/fileio/base.py

```diff
@@ -0,0 +1,166 @@
+"""
+Implementation of the base classes for the ORSO header.
+"""
+
+# author: Andrew R. McCluskey (arm61)
+
+import yaml
+
+
+def noop(self, *args, **kw):
+    pass
+
+
+yaml.emitter.Emitter.process_tag = noop
+
+
+def _repr(class_to_represent):
+    """
+    The representation object for all the Header sub-classes. This returns a
+    string in a yaml format which will be ORSO compatible.
+
+    Args:
+        class_to_represent (:py:class:`object`): The class to be represented.
+
+    Returns:
+        (:py:attr:`str`): A string representation.
+    """
+    cleaned_data = dict(
+        (k, v) for (k, v) in class_to_represent.__dict__.items(
+        ) if v is not None
+    )
+    return yaml.dump(cleaned_data, sort_keys=False)
+
+
+class Header:
+    """
+    The super class for all of the __repr__ items in the orso module
+    """
+
+    def __repr__(self):
+        """
+        The string representation for the Header class objects.
+        """
+        return _repr(self)
+
+
+class ValueScalar(Header):
```

Comment:
I'm not sure why we need such a large class structure. In principle a dictionary or derived object would be sufficient that then just checks against the schema file. In this way we only have to update a scheme and not the code here.
For me this would be quite a lot of additional overhead when using the package to put all my values first in one of these classes upon writing and extract the values from the classes when reading. It seems simpler to just supply a full dictionary and get an error message if it does not follow the schema. (Warning for recommendations, error for missing data or wrong entries.)

In case you like to stick to custom classes for everything, any kind of checks should be methods of the associated classes. Then the classes also need a facility to check for validity in certain entries against the schema. (e.g. if values and value ranges are allowed).

Also keep in mind that the library should be used by other programs, so it would be good to make sure resulting output is available in standard types. (str, float, int, list, dict and ndarray)

Reply:
My motivation for the class structure was two fold: 
- making checking easier
- making documentation clearer/easier

I am very open to changing my mind, but this was how I designed the amor code so I was going off of that. 

There is no reason we can't have a helper function(s) that consumes/produces a dictionary for the full header.